src/pages/ScholarshipForm.jsx

```diff
@@ -42,26 +133,54 @@ function ScholarshipForm() {
     <Container maxWidth="md">
       <form>
         <h1>Submit a Scholarship</h1>
-        {requiredTextField('name', 'Scholarship Name')}
+        <div>
+          {requiredTextField(
+            'name',
+            'Scholarship Name',
+            formFieldErrors.name,
+            'Please enter a name'
+          )}
+        </div>
         <DatePicker
           id="deadline"
           label="Deadline"
+          error={formFieldErrors.deadline}
+          helperText="Please enter a valid date"
           value={formFieldStates.deadline}
           onChange={(date) =>
             setFormFieldStates({ ...formFieldStates, deadline: date })
           }
         />
-        {requiredTextField('description', 'Description')}
-        {requiredTextField('website', 'Website')}
+        <div>
+          {requiredTextField(
+            'description',
+            'Description',
+            formFieldErrors.description,
+            'Please enter a description',
+            true
+          )}
+        </div>
+        <div>
+          {requiredTextField(
+            'website',
+            'Website',
+            formFieldErrors.website,
+            'Please enter a valid website'
+          )}
```

Comment:
Another field you could accept is `type`, e.g. so that [`type="url"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/url) can be passed instead of having our own URL validation. `Input` is one of the components `TextField` [inherits from](https://material-ui.com/api/text-field/#advanced-configuration)

Reply:
Adding to this, I think the `<form>` component on submission will actually validate a valid URL is passed (might want to double check). But for that you need to set the submit button's `type` to `"submit"`

Reply:
#209 can explore this